contrib/PyCuAmpcor/src/cuAmpcorChunk.cu

```diff
@@ -65,29 +65,31 @@ void cuAmpcorChunk::run(int idxDown_, int idxAcross_)
     // 41 x 41, if halfsearchrange=20
     cuArraysMaxloc2D(r_corrBatchRaw, offsetInit, r_maxval, stream);
 
-    // Estimation of statistics
-    // Extraction of correlation surface around the peak
+    // estimate variance
+    cuEstimateVariance(r_referenceBatchRaw->size, r_corrBatchRaw, offsetInit, r_maxval, r_covValue, stream);
+
+    // estimate SNR
+    // step1: extraction of correlation surface around the peak
     cuArraysCopyExtractCorr(r_corrBatchRaw, r_corrBatchRawZoomIn, i_corrBatchZoomInValid, offsetInit, stream);
 
-    // Summation of correlation and data point values
+    // step2: summation of correlation and data point values
     cuArraysSumCorr(r_corrBatchRawZoomIn, i_corrBatchZoomInValid, r_corrBatchSum, i_corrBatchValidCount, stream);
 
 #ifdef CUAMPCOR_DEBUG
+    r_maxval->outputToFile("r_maxval", stream);
+    r_corrBatchRawZoomIn->outputToFile("r_corrBatchRawStatZoomIn", stream);
     i_corrBatchZoomInValid->outputToFile("i_corrBatchZoomInValid", stream);
     r_corrBatchSum->outputToFile("r_corrBatchSum", stream);
+    i_corrBatchValidCount->outputToFile("i_corrBatchValidCount", stream);
 #endif
 
-    // SNR
+    // step3: divide the peak value by the mean of surrounding values
     cuEstimateSnr(r_corrBatchSum, i_corrBatchValidCount, r_maxval, r_snrValue, stream);
 
-    // Variance
-    cuEstimateVariance(r_corrBatchRaw, offsetInit, r_maxval, r_covValue, stream);
-
```

Comment:
need to call it with the new definition

Reply:
This deleted code of the old version. To being consistent with ampcor.F, I put variance estimation before snr estimation. The variance estimate is at line 69 in the new code here.